buildenv/jenkins/JenkinsfileBase

```diff
@@ -461,13 +461,18 @@ def uploadToArtifactory() {
 				]
 			}"""
 
-		def buildInfo = Artifactory.newBuildInfo()
-		// if ARTIFACTORY_NUM_ARTIFACTS is not set, set to keep 20 build artifacts
-		def numArtifacts = params.ARTIFACTORY_NUM_ARTIFACTS ? params.ARTIFACTORY_NUM_ARTIFACTS : 20
-
-		buildInfo.retention maxBuilds: numArtifacts, deleteBuildArtifacts: true
-		server.upload spec: uploadSpec, buildInfo: buildInfo
-		server.publishBuildInfo buildInfo
+		if (params.ARTIFACTORY_VERSION == 'ent') {
```

Comment:
- Could we make `ent` to be the default value? The current code force all existing jobs that use artifactory to provide an extra parameter - ARTIFACTORY_VERSION.
- Do we have a cleanup story for `oss`?

Reply:
We can, up to you. That is how I structured the OpenJ9 version of the `if`, but that doesn't mean we have to do the same thing for test. If you prefer I can do `if oss, else`.
For the cleanup, Sam has a pr eclipse/openj9#4996 which I have yet to review.

Reply:
On second thought, is it possible to set `ARTIFACTORY_VERSION` based on `ARTIFACTORY_SERVER` value? As we set the Artifactory `Server ID` in [Jenkins artifacoty plugin](https://ci.eclipse.org/openj9/configure), so we can decide on how to name the server. Other alternatives could be finding artifactory version through artifactory API or using try catch to see if we can get `buildInfo`. But the first option seems simpler.

It would be great if we could do it without introducing a new parameter as we are already experiencing parameters explosion https://github.com/AdoptOpenJDK/openjdk-tests/issues/902

Reply:
Maybe this will help (isArtifactoryPro)
https://javadoc.jenkins.io/plugin/artifactory/org/jfrog/hudson/ArtifactoryServer.html#isArtifactoryPro-org.jfrog.hudson.DeployerOverrider-hudson.model.Item-
Will follow up tomorrow.

Reply:
Well that didn't work. The server object created is actually a different class org.jfrog.hudson.pipeline.common.types.ArtifactoryServer
not 
org.jfrog.hudson.ArtifactoryServer
Which doesn't have a method we could use to get the version afaict. I looked through the Doc and cannot find a way to get what we need.

The alternative would be to do a REST API call over https.
https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-VersionandAdd-onsinformation
For OSS version returns
```
{
  "version" : "6.5.2",
  "revision" : "60502900",
  "addons" : [ ],
  "license" : "Artifactory OSS"
}
```
The Pro version has a large license key in the license field. So I could do the call and do `if OSS <1>, else <2>`.

Reply:
Have a workaround that curls the API. Will test tomorrow. Also opened a feature request on the plugin
https://issues.jenkins-ci.org/browse/JENKINS-56512

Reply:
Thanks a lot for trying different options, Adam. :)

Reply:
I was trying to get the ART credentials via the `server` object
`server.getCredentialsId()` but that doesn't seem to be working (always returns `null`).
I opened https://issues.jenkins-ci.org/browse/JENKINS-56531 to address that issue. As a workaround, for our workaround, I thought about using `CUSTOMIZED_SDK_URL_CREDENTIAL_ID` since we already pass that for the get.sh. The problem with that would be, for example, a user running a grinder on taas, pulling an sdk from adopt. The get.sh wouldn't require the creds but the Artifactory upload step would.

I'm going to see if we can try/catch the upload to always do buildinfo and if it fails, upload without it. 
```
try {
	server.upload spec: uploadSpec, buildInfo: buildInfo
	server.publishBuildInfo buildInfo
} catch {
	server.upload spec: uploadSpec
}
```
Will update once I test it. hopefully the combined try/catch doesn't add alot of extra time.

Reply:
I'm glad we tried the try/catch. The original code is working now. I can publish buildInfo to the OSS server. The artifact properties and build retention data is basically ignored but it doesn't fail. I'm not sure why it failed when I attempted it last week. I will confirm on the test side and if it works I will close this PR.